### PR TITLE
refactor: simplify <webview> event dispatch

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
We no longer need special handling for emitting `ipc-message`.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes

#### Release Notes
Notes: none

### CODE REVIEW

**Review by zcbenz — APPROVED** [review]

File: lib/browser/guest-view-manager.ts [hunk 1]
```diff
@@ -136,27 +136,30 @@ const createGuest = function (embedder: Electron.WebContents, embedderFrameId: n
...
+      sendToEmbedder(IPC_MESSAGES.GUEST_VIEW_INTERNAL_DISPATCH_EVENT, event, makeProps(event, args));
+    });
   }
 
   guest.on('new-window', function (event, url, frameName, disposition, options, additionalFeatures, referrer) {
-    sendToEmbedder(IPC_MESSAGES.GUEST_VIEW_INTERNAL_DISPATCH_EVENT, 'new-window', url,
-      frameName, disposition, sanitizeOptionsForGuest(options),
-      additionalFeatures, referrer);
+    sendToEmbedder(IPC_MESSAGES.GUEST_VIEW_INTERNAL_DISPATCH_EVENT, 'new-window', {
+      url, frameName, disposition, options: sanitizeOptionsForGuest(options), additionalFeatures, referrer
```
Comment (nornagon) [inline lib/browser/guest-view-manager.ts:156]: nit: let's break this apart a bit for readability
```suggestion
      url,
      frameName,
      disposition,
      options: sanitizeOptionsForGuest(options),
      additionalFeatures,
      referrer
```

**Review by nornagon — APPROVED** [review]

File: lib/browser/guest-view-manager.ts [hunk 2]
```diff
@@ -136,27 +136,33 @@ const createGuest = function (embedder: Electron.WebContents, embedderFrameId: n
...
+    guest.on(event as any, function (_, ...args: any[]) {
+      sendToEmbedder(IPC_MESSAGES.GUEST_VIEW_INTERNAL_DISPATCH_EVENT, event, makeProps(event, args));
+    });
   }
 
-  guest.on('new-window', function (event, url, frameName, disposition, options, additionalFeatures, referrer) {
-    sendToEmbedder(IPC_MESSAGES.GUEST_VIEW_INTERNAL_DISPATCH_EVENT, 'new-window', url,
-      frameName, disposition, sanitizeOptionsForGuest(options),
-      additionalFeatures, referrer);
+  guest.on('new-window', function (event, url, frameName, disposition, options) {
```
Comment (miniak) [inline lib/browser/guest-view-manager.ts:154]: removing `additionalFeatures`, `referrer` here as they were not mapped to `'new-window'` event props before, will  be fixed in a separate PR.

**Review by jkleinsc — APPROVED** [review]
